fix: fix JWT signing key error

james0248 opened the pull request.

james0248 committed:
build: update dependency, add properties

james0248 committed:
fix: fix JWT signing key to Key type

yeonsumia commented on src/main/kotlin/com/wafflestudio/msns/domain/user/dto/UserRequest.kt:
```diff
@@ -6,44 +6,23 @@ import javax.validation.constraints.NotNull
-
-        @field:NotBlank
-        val birth: String
+        val countryCode: String,
+        @field:NotBlank val password: String,
+        @field:NotBlank val name: String,
+        @field:NotBlank val username: String,
+        @field:NotBlank val birthDate: String
```
이와 같은 refactoring이 필요한가요?

james0248 replied:
이거도 컨벤션 차이일것 같습니다. 개인적으로 dto는 라인수 굳이 많이 잡아먹을 필요없다고 생각해서 수정했어요. 필드 사이에 개행 하시는 이유가 있을까요?

yeonsumia replied:
다른 dto 파일들도 이처럼 반영이 안 된 것 같은데, 한 가지 방식으로 통일되면 좋겠습니다.

yeonsumia commented on src/main/kotlin/com/wafflestudio/msns/domain/user/model/User.kt:
```diff
@@ -6,52 +6,60 @@ import java.time.LocalDate
 
-    @Column(columnDefinition = "BINARY(16)")
-    val streamId: UUID,
+    @Column(name = "introduction")
+    var introduction: String? = null,
```
이 부분은 어떤 이유로 수정하신 건가요?

james0248 replied:
이건 다시 revert하겠습니다. 그냥 빈 스트링이 더 나아보이네요

yeonsumia commented on src/main/kotlin/com/wafflestudio/msns/domain/user/dto/UserResponse.kt:
```diff
@@ -24,32 +24,30 @@ class UserResponse {
         val username: String,
-        val image: String,
-        val introduction: String,
+        val profileImageUrl: String?,
```
이러한 처리는 좋습니다.

yeonsumia commented on src/main/kotlin/com/wafflestudio/msns/global/auth/service/AuthService.kt:
```diff
@@ -55,34 +52,33 @@ class AuthService(
 
         val verificationToken = verificationTokenRepository.findByEmail(email)
-            ?.also { if (!it.verification) throw UnauthorizedVerificationTokenException("email is unauthorized.") }
+            ?.also { if (!it.isVerified) throw UnauthorizedVerificationTokenException("email unauthorized") }
```
전반적으로 적절하게 변수명 잘 수정하신 것 같습니다.

yeonsumia commented on src/main/kotlin/com/wafflestudio/msns/global/mail/service/MailService.kt:
```diff
@@ -11,9 +11,7 @@ import javax.mail.internet.InternetAddress
-class MailService(
-    private val emailSender: JavaMailSender,
-) {
+class MailService(private val emailSender: JavaMailSender) {
```
불필요한 수정으로 보입니다.

james0248 replied:
컨벤션이 정해져 있으면 그쪽으로 맞춰서 바꾸겠습니다. 한줄이여도 무조건 개행하는것으로 할까요?

yeonsumia commented on src/main/kotlin/com/wafflestudio/msns/domain/user/model/User.kt:
```diff
@@ -6,52 +6,60 @@ import java.time.LocalDate
     val likes: MutableList<Like> = mutableListOf(),
 
-) : BaseTimeEntity()
+    ) : BaseTimeEntity()
```
불필요한 수정으로 보입니다.

yeonsumia commented on src/main/kotlin/com/wafflestudio/msns/global/auth/service/AuthService.kt:
```diff
@@ -39,9 +37,8 @@ class AuthService(
-    fun signUpEmail(emailRequest: AuthRequest.JoinEmail): Boolean {
+    fun verifyEmail(emailRequest: AuthRequest.VerifyEmail): Boolean {
         val email = emailRequest.email
-        if (!isEmailValid(email)) throw InvalidEmailFormException("Invalid Email Form")
```
이 부분 삭제한 이유가 뭔가요?

james0248 replied:
DTO 단에서 `@Email` 어너테이션으로 검증하면 굳이 서비스 단에서 다시 검증할 필요 없다고 생각했습니다.

yeonsumia commented on src/main/resources/application.yml:
```diff
@@ -22,13 +22,28 @@ spring:
     hibernate:
       ddl-auto: create
     show-sql: true
+  mail:
+    host: smtp.gmail.com
+    port: 587
+    username: auth.feelin@gmail.com
+    password: cubvolcblqwqorbv
+    properties:
+      mail:
+        smtp:
+          auth: true
+          starttls:
+            enable: true
+            required: true
+  jwt:
+    secret: feelin-server-jwt-secret-key-2020
+    expiration: 86400000
```
secret 정보를 이렇게 공개적으로 다루시면 안됩니다.

james0248 replied:
JWT secret도 같이 코드 상에서 지워야할것 같아요

yeonsumia commented on src/main/kotlin/com/wafflestudio/msns/global/auth/jwt/JwtTokenProvider.kt:
```diff
@@ -50,11 +51,12 @@ class JwtTokenProvider(
                 JWT.REFRESH -> jwtRefreshExpirationInMs
             }
         )
+        val key: Key = Keys.hmacShaKeyFor(jwtSecretKey.toByteArray())
```
plain string 관련된 부분이 이것인가요?

james0248 replied:
넵 맞습니다. 라이브러리 업그레이드 하면서 key 길이도 256bit 이상이여야 해서 secret도 변경 같이 했습니다

yeonsumia reviewed (changes requested):
전반적으로 변수명 및 적절한 함수 대체로 refactoring이 되었지만, 불필요한 수정도 있습니다. 또한 secret을 코드에서 공개적으로 사용하지 않습니다.

james0248 committed:
fix: refactored due to review

- change User introduction to non-nullable
- isVerified to verified
- line break in constructor
- remove secrets

yeonsumia commented on src/main/kotlin/com/wafflestudio/msns/global/auth/dto/AuthRequest.kt:
```diff
@@ -1,11 +1,14 @@
-    data class JoinEmail(
-        val email: String,
+    data class VerifyEmail(
+        @field:NotBlank @field:Email val email: String,
```
여기도 개행 방식 통일 부탁합니다.

james0248 replied:
어떤 부분을 통일해달라는 건지 이해가 잘 안갑니다. 다른 곳과 동일한 것 같아요

yeonsumia commented on src/main/resources/application.yml:
```diff
@@ -22,13 +22,28 @@ spring:
     hibernate:
       ddl-auto: create
     show-sql: true
+  mail:
+    host: smtp.gmail.com
+    port: 587
+    username: ~
+    password: ~
+    properties:
+      mail:
+        smtp:
+          auth: true
+          starttls:
+            enable: true
+            required: true
+  jwt:
+    secret: ~
+    expiration: 86400000
```
이 부분은 서버에서 적용할 테니 없어도 될 듯 합니다. 계정 비밀번호는 바꾸었고, secret 정보는 dm으로 보내주길 바랍니다.

james0248 committed:
build: change DB name to match application.yml

james0248 commented:
@yeonsumia 
- DTO annotation 관련 포매팅은 이 PR에서 전부 바꾸지 말고 별도의 style PR로 분리하는 것이 좋아보입니다.
- application.yml에서 로컬 환경에서 세팅이 어려워 일부 property를 변경했습니다. (DB 이름, DB user password) 이에 맞춰 docker compose 파일도 변경했는데 이거면 충분할까요?
- 일단 secret 정보는 마스킹을 해놨는데 이런 설정은 어떤식으로 주입하시는걸까요?
